planner/optimize.go

```diff
@@ -32,6 +39,26 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 		return fp, nil
 	}
 
+	var oriHint *bindinfo.HintsSet
+	if stmtNode, ok := node.(ast.StmtNode); ok {
+		node, oriHint = addHint(sctx, stmtNode)
+	}
+	plan, err := optimize(ctx, sctx, node, is)
```

Comment:
What about
```suggestion
	if oriHint == nil {
		plan, err := optimize(ctx, sctx, node, is)
		return plan, err
	}
```
to avoid unnecessary optimize.

Reply:
Why? We need to optimize even if `oriHint` is not nil.

Reply:
Yes, but we can avoid executing twice optimize when `oriHint != nil`.

Reply:
But we need to optimize twice if `oriHint` is not nil because the hint maybe outdated.

Reply:
If oriHint is out of date. Why we need the first optimize?

Reply:
Because you have to try optimize to know if it is outdated.